### PR TITLE
Company ID GUI tool

### PR DESCRIPTION
**Pull request by lassemoldrup**
## Purpose

Addresses #109.

## Checklist

- [x] My code follows the style of this project.
- [x] The code compiles without warnings.
- [x] I have performed a self-review of the changes.
- [x] I have documented my code, in particular the intent of the
      hard-to-understand areas.
- [ ] (If necessary) I have updated the CHANGELOG.

### CODE REVIEW

File: company-id-gui/src-tauri/src/main.rs [hunk 1]
```diff
@@ -0,0 +1,721 @@
...
+        ip_index,
+        ..
+    } = state.wallet_context(&seedphrase).await?;
+    let net = wallet.net;
+
+    let acc_data = get_account_keys(&wallet, ip_index, id_index, account.acc_index)?;
+    let acc_keys = AccountKeys {
+        keys:      [(CredentialIndex::from(0), acc_data)].into(),
+        threshold: AccountThreshold::ONE,
+    };
```
Comment (abizjak) [inline company-id-gui/src-tauri/src/main.rs:526]:
```suggestion
    let acc_keys: AccountKeys = acc_data.into();
```

File: company-id-gui/src-tauri/src/main.rs [hunk 2]
```diff
@@ -0,0 +1,721 @@
...
+    wallet: &ConcordiumHdWallet,
+    ip_index: u32,
+    id_index: u32,
+    acc_index: u8,
+) -> Result<CredentialData, Error> {
+    let secret = wallet
+        .get_account_signing_key(ip_index, id_index, acc_index as u32)
+        .context("Failed to get account signing key.")?;
+    let mut keys = std::collections::BTreeMap::new();
+    let public = ed25519_dalek::PublicKey::from(&secret);
```
Comment (abizjak) [inline company-id-gui/src-tauri/src/main.rs:498]:
```suggestion
    let public = (&secret).into();
```

And you can remove the dependency on ed25519_dalek.

**Review by abizjak — COMMENTED** [review]
Why do we need a vite.svg and tauri.svg in the repository? And the stuff under icons .

Why do we have the empty assets folder in the repository? Why is there an empty App.scss file?

File: company-id-gui/.vscode/extensions.json [hunk 3]
```diff
@@ -0,0 +1,3 @@
+{
+  "recommendations": ["tauri-apps.tauri-vscode", "rust-lang.rust-analyzer"]
+}
```
[the remark this answers is not in the document]
Comment (lassemoldrup) [reply company-id-gui/.vscode/extensions.json:3]: The file is specifically allowed in the `.gitignore` from our dapp template.

File: company-id-gui/.editorconfig [hunk 4]
```diff
@@ -0,0 +1,9 @@
+root = true
+
+[*]
+indent_style = space
+indent_size = 4
+end_of_line = lf
+charset = utf-8
+trim_trailing_whitespace = true
+insert_final_newline = true
```
[the remark this answers is not in the document]
Comment (lassemoldrup) [reply company-id-gui/.editorconfig:9]: This is from our dApp template.

File: company-id-gui/.vscode/extensions.json (hunk 3 above)
[the remark this answers is not in the document]
Comment (abizjak) [reply company-id-gui/.vscode/extensions.json]: That's a mistake in that template then.

File: company-id-gui/src/components/App/App.tsx [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+    const [nodeURLError, setNodeURLError] = useState(null as string | null);
+    const [nodeURL, setNodeURL] = useState(null as string | null);
+    const [isConnecting, setIsConnecting] = useState(false);
+
+    const defaultNodeURL =
+        network === Network.Testnet
+            ? 'http://node.testnet.concordium.com:20000'
+            : 'https://grpc.mainnet.concordium.software:20000';
+
+    const actualNodeURL = nodeURL ?? defaultNodeURL;
```
[the remark this answers is not in the document]
Comment (lassemoldrup) [reply company-id-gui/src/components/App/App.tsx:27]: Yes, if `nodeURL` is set (by editing the URL) the render function will run again, thereby updating `actualNodeURL`.

**Review by abizjak — APPROVED** [review]

**Pull request merged**